src/currencyConversion/BitmamaAPI.ts

```diff
@@ -1,21 +1,42 @@
 import { RESTDataSource } from 'apollo-datasource-rest'
 import BigNumber from 'bignumber.js'
 import { logger } from '../logger'
-import { BITMAMA_API } from '../config'
 
 interface BitmamaAPIResult {
   buy: number
   sell: number
 }
 
 // ttl in seconds!
-const MIN_TTL = 12 * 3600 // 12 hours
+const MIN_TTL = 10 * 60 // 10 mintues
+
+/**
+ * API Client for Bitmama's exchange rate API.
+ *
+ * Two types of caching are used:
+ * - apollo-datasource-rest's built-in expiring cache is used to save a roundtrip and regulate our API usage
+ * - a local, non-expiring cache is used as a fallback in case requests fail
+ */
 export default class BitmamaAPI extends RESTDataSource {
-  private rateCache: Map<string, BigNumber>
-  constructor() {
+  private rateCache: Map<
+    string,
+    {
+      rate: BigNumber
+      timestamp: number
+    }
+  >
+  constructor(baseUrl: string) {
     super()
-    this.baseURL = BITMAMA_API
-    this.rateCache = new Map<string, BigNumber>()
+    this.baseURL = baseUrl
+    // Should be safe to not have an eviction policy here since we only cache NGN/USD rates
+    // and even eif we used all 200+ currencies, the memory usage would be negligible
```

Comment:
```suggestion
    // and even if we used all combinations of all 200ish currencies, the memory usage would be acceptable
```
idk if I'd call 40k records-- which is the worst case-- negligible

Reply:
also I switched from "200+" to "200ish" because the purpose of the comment was to estimate an upper bound. When calculating an upper bound, it doesn't work to use lower bounds (like "at least 200") for the inputs.